lib/scnnr/recognition.rb

```diff
@@ -24,7 +25,13 @@ def error?
     end
 
     def to_h
-      { 'id' => self.id, 'objects' => self.objects.map(&:to_h), 'state' => self.state.to_s }
+      hash = {
+        'id' => self.id,
+        'image' => self.image&.to_h,
+        'objects' => self.objects.map(&:to_h),
+        'state' => self.state.to_s,
+      }
+      hash.reject { |_, value| value.nil? }
```

Comment:
Sorry, I reconsidered this and I think  including `{ image: nil }` is not bad, because `Scnnr::Recognition` has `image` instance as `nil` and it is unnecessary to exclude it. 

So it is no problem that `to_h` method returns `{ 'id' => self.id, 'image' => self.image&.to_h, 'objects' => self.objects.map(&:to_h), 'state' => self.state.to_s }` . 
What do you think?

Reply:
I think it make sense. But in my opinion it’s better to remove it since there’s nothing to do with it and also if we return nil on image it will cause some confusion since they know that they pass an image, so I think it’s better if we remove it.

Reply:
Thank you for your opinion!
 I think `{ image: nil }` means that `Recognition` instance does not have `image` and it make sense. 
Besides, our API documentation describes about `image`’s existance, so I think it is not confusing :slightly_smiling_face: What made you confused?

Reply:
I'm just thinking that it will cause confusion since they passed an image but on the result they are getting nil.

Reply:
And also if we will do this we should also change this one `@image = Image.new(attrs['image']) if attrs['image']`, so they know that there is an image instance even if the value is nil. What do you think @danimal141 ?

Reply:
> since they passed an image but on the result they are getting nil

I can understand what you mean, but It is API specification, so API client should act according to the specification. If it is bad, We may need to reconsider both API and API client, which will be other task.


> And also if we will do this we should also change this one @image = Image.new(attrs['image']) if attrs['image'], so they know that there is an image instance even if the value is nil.

Sorry, I cannot understand your opinion correctly... what do you mean? Can you show me the example code or something?

Reply:
> And also if we will do this we should also change this one @image = Image.new(attrs['image']) if attrs['image'], so they know that there is an image instance even if the value is nil.

Forgot this one, I just realized it doesn't make sense.

I'll send an update for my PR.